maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -562,6 +566,21 @@ private void commands(CliRequest cliRequest) {
     // Maybe it's better to move some of those methods to separate class (SoC).
     void properties(CliRequest cliRequest) {
         populateProperties(cliRequest.commandLine, cliRequest.systemProperties, cliRequest.userProperties);
+
+        // now that we have properties, interpolate all arguments
+        BasicInterpolator interpolator = createInterpolator(cliRequest.systemProperties, cliRequest.userProperties);
+        CommandLine.Builder commandLineBuilder = new CommandLine.Builder();
+        cliRequest.commandLine.getArgList().stream()
+                .map(s -> {
+                    try {
+                        return interpolator.interpolate(s);
+                    } catch (InterpolationException e) {
+                        throw new RuntimeException("Unable to interpolate", e);
```

Comment:
This should not be a runtime exception since it comes from input external to the program. InterpolationException might be OK, or perhaps a different checked exception. 

If this means, the code doesn;t use lambdas, that's fine.